geode-core/src/main/java/org/apache/geode/internal/security/shiro/JMXShiroAuthenticator.java

```diff
@@ -48,13 +48,16 @@ public JMXShiroAuthenticator(SecurityService securityService) {
 
   @Override
   public Subject authenticate(Object credentials) {
-    String username = null;
+    String username;
     Properties credProps = new Properties();
     if (credentials instanceof String[]) {
       final String[] aCredentials = (String[]) credentials;
       username = aCredentials[0];
       credProps.setProperty(ResourceConstants.USER_NAME, aCredentials[0]);
       credProps.setProperty(ResourceConstants.PASSWORD, aCredentials[1]);
+    } else if (credentials instanceof String) {
+      username = "";
```

Comment:
this probably is not good. maybe use a unique string based on the token would be nice.

Reply:
I addressed this by setting username to "token-" + credential.hashCode()